lib/llm/src/grpc/service/kserve.rs

```diff
@@ -471,42 +534,50 @@ impl GrpcInferenceService for KserveService {
             .find(|card| request_model_name == &card.display_name)
         {
             if card.model_type.supports_tensor() {
-                if let Some(tensor_model_config) = card.runtime_config.tensor_model_config.as_ref()
-                {
-                    let model_config = ModelConfig {
-                        name: tensor_model_config.name.clone(),
-                        platform: "dynamo".to_string(),
-                        backend: "dynamo".to_string(),
-                        input: tensor_model_config
-                            .inputs
-                            .iter()
-                            .map(|input| ModelInput {
-                                name: input.name.clone(),
-                                data_type: input.data_type.to_kserve(),
-                                dims: input.shape.clone(),
-                                ..Default::default()
-                            })
-                            .collect(),
-                        output: tensor_model_config
-                            .outputs
-                            .iter()
-                            .map(|output| ModelOutput {
-                                name: output.name.clone(),
-                                data_type: output.data_type.to_kserve(),
-                                dims: output.shape.clone(),
-                                ..Default::default()
-                            })
-                            .collect(),
-                        ..Default::default()
-                    };
-                    return Ok(Response::new(ModelConfigResponse {
-                        config: Some(model_config.clone()),
-                    }));
+                let config = Config::from_runtime_config(&card.runtime_config).map_err(|e| {
+                    Status::invalid_argument(format!(
+                        "Model '{}' has type Tensor but: {}",
+                        request_model_name, e
+                    ))
+                })?;
+                match config {
+                    Config::Triton(model_config) => {
+                        return Ok(Response::new(ModelConfigResponse {
+                            config: Some(model_config),
+                        }));
+                    }
+                    Config::Dynamo(tensor_model_config) => {
+                        let model_config = ModelConfig {
+                            name: tensor_model_config.name.clone(),
+                            platform: "dynamo".to_string(),
```

Comment:
Can you writeup some documentation on the kserve frontend before you go on PTO? Separate PR is OK.

High level things I'd want to see in something like `docs/frontend/kserve.md`:
1. ModelInput.Tensor behavior / motivation
2. ModelInput.text/tokens behavior and how we support conversion
3. Triton Model Config compatibiilty/interaction
4. Reference to example tensor echo worker
5. Reference to python bindings doc: https://github.com/ai-dynamo/dynamo/blob/main/lib/bindings/python/examples/kserve_grpc_service/README.md
6. Limitations
    - implications (if any) of hard-coded `"dynamo"`  platform/backend I see here
    - implications (if any) of hard-coded "version=1" I see here
    - ...